mautic/mautic
# Fast rendering segment list

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ❌ 
| New feature/enhancement? (use the a.x branch)      | ✅ 
| Deprecations?                          | ❌ 
| BC breaks? (use the c.x branch)        | ❌ 
| Automated tests included?              | ✅  
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
Listing segments with large contacts is slow. A single segment with several millions of contacts causes the load time of this page to raise above 6 seconds.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Build segment with thousands to millions of contacts
3. Go to segment listing page




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11071"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
